rewrite-java/src/main/java/org/openrewrite/java/JavaParser.java

```diff
@@ -73,28 +73,32 @@ static List<Path> runtimeClasspath() {
      * matching jars can be found.
      */
     static List<Path> dependenciesFromClasspath(String... artifactNames) {
-        Map<String, Pattern> artifactNamePatterns = Arrays.stream(artifactNames)
-                .collect(toMap(Function.identity(), name -> Pattern.compile(name + "-.*?\\.jar$")));
-
-        Set<String> foundArtifacts = new HashSet<>();
-        List<Path> artifacts = new ArrayList<>();
         List<URI> runtimeClasspath = new ClassGraph().getClasspathURIs();
-        for (URI cpEntry : runtimeClasspath) {
-            for (Map.Entry<String, Pattern> artifactNamePattern : artifactNamePatterns.entrySet()) {
-                if (artifactNamePattern.getValue().matcher(cpEntry.toString()).find()) {
+        List<Path> artifacts = new ArrayList<>(artifactNames.length);
+        List<String> missingArtifactNames = new ArrayList<>(artifactNames.length);
+        for (String artifactName : artifactNames) {
+            Pattern jarPattern = Pattern.compile(artifactName + "-.*?\\.jar$");
+            // In a multiproject IDE classpath, some classpath entries aren't jars
+            Pattern explodedPattern = Pattern.compile("/" + artifactName + "/");
+            boolean lacking = true;
+            for (URI cpEntry : runtimeClasspath) {
+                String cpEntryString = cpEntry.toString();
+                if (jarPattern.matcher(cpEntryString).find()
+                        || (explodedPattern.matcher(cpEntryString).find())
+                        && Paths.get(cpEntry).toFile().isDirectory()) {
                     artifacts.add(Paths.get(cpEntry));
-                    foundArtifacts.add(artifactNamePattern.getKey());
+                    lacking = false;
+                    break;
```

Comment:
This is causing the tests to break for artifacts like "junit", because the old+new jarPattern is unsafe.
With "junit-foo", "junit" and "junit-bar" in the classpath, looking for "junit" will only yield "junit-foo", not "junit"